dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatMessageBase.cs

```diff
@@ -34,6 +34,6 @@ protected ChatMessageBase(AuthorRole role, string content, IDictionary<string, s
     {
         this.Role = role;
         this.Content = content;
-        this.AdditionalProperties = additionalProperties;
+        this.AdditionalProperties = additionalProperties ?? new Dictionary<string, string>();
```

Comment:
This is going to allocate a new dictionary for every chat message even if the dictionary is never needed.  It should instead be done lazily as part of the getter.